### PR TITLE
Bump supported versions for 3.1 release

### PR DESCRIPTION
**Pull request by vbelolapotkov**
#### Changes proposed in this Pull Request

* Bump the supported versions for the 3.1 release.
* Bump min required WC Version to 4.4. ✅ Discussed with Architecture guild paJDYF-2Dr-p2#comment-8197
* Update CI matrix to cover latest versions of WC. 
* Run edge unit tests with PHP 7.1 to avoid files loading issues. To be fixed in #2970

Per L-2 policy, this PR bumps the supported versions of WooCommerce to 5.5-5.7.

WC Pay doesn't work properly on WC 4.0 - 4.3 (p1632307719018700/1632161800.465400-slack-CGGCLBN58) hence bumping min version to 4.4 according to pending improvements in issue #1775.

WP version bump is not needed.

#### Testing instructions

* Check CI matrix changes.
* All checks should pass.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by tpaksu — APPROVED** [review]
LGTM, you can 🚢 it.

File: .github/workflows/e2e-test.yml
Comment (htdat) [inline .github/workflows/e2e-test.yml:31]: Should we have `5.6.0` too?

File: .github/workflows/compatibility.yml
Comment (htdat) [inline .github/workflows/compatibility.yml:21]: Since we no longer run PHPUnit with PHP 7.0, I think this line below should be removed too: 

```bash
- run: if [[ "${{ matrix.php }}" == '7.0' ]]; then wget https://phar.phpunit.de/phpunit-6.5.14.phar && mv phpunit-6.5.14.phar phpunit.phar; fi;
```

File: .github/workflows/e2e-test.yml
Comment (kalessil) [reply .github/workflows/e2e-test.yml:31]: Not really, I'd suggest even skipping `'5.5.0'` . @vbelolapotkov what was the idea behind adding both `5.5.0` and `5.7.0` here?

File: .github/workflows/compatibility.yml
Comment (kalessil) [reply .github/workflows/compatibility.yml:21]: +1 here

**Review by kalessil — CHANGES_REQUESTED** [review]
Overall looks great! Left a couple of comments only.

File: .github/workflows/compatibility.yml
Comment (vbelolapotkov) [reply .github/workflows/compatibility.yml:21]: I think it's ok for now, but we'd need to put it back once/if we solve plugin loading issue for PHP 7 & PHP Unit 6.5.14

File: .github/workflows/e2e-test.yml
Comment (vbelolapotkov) [reply .github/workflows/e2e-test.yml:31]: The idea was to test both edges of L-2 support range. So the full range is:
Min WC | L-2 | L(atest) | beta

**Comment by vbelolapotkov** [comment]
Putting needs author reply label to update the PR once we aligned on WC min version bump. @htdat will continue work on this PR.

File: .github/workflows/compatibility.yml
Comment (htdat) [reply .github/workflows/compatibility.yml]: Commented out this line, and added todos for this line. Ref: 0f234a3

File: .github/workflows/e2e-test.yml
Comment (htdat) [reply .github/workflows/e2e-test.yml:32]: > Min WC | L-2 | L(atest) | beta

Thanks for the explanation. If so, this change makes sense.
I've added a comment to explain why we keep this, which would be helpful for the next maintainers. Ref 16148f3

**Comment by htdat** [comment]
Can this PR have another review? Some changes: 

1, Discussed with Architecture guild. paJDYF-2Dr-p2#comment-8197:

- We will go ahead with Bump WC version from 4.0 to 4.4 
- And continue loading WCPay if the account is currently connected on their sit, which is addressed in this PR https://github.com/Automattic/woocommerce-payments/pull/3010

2, I did not make many changes, mostly adding up comments/TODOs and commenting out the line grabbing PHPUnit 6.5.14 for PHP 7.0.

Note: this test E2E tests / E2E tests (beta, latest, 7.4) failing seems irrelevant.

**Comment by htdat** [comment]
Note: all E2E tests are falling due to one same test and not relevant to this PR. Asking here p1632983716124200-slack-CGGCLBN58

```
Summary of all failing tests
FAIL tests/e2e/specs/shopper/shopper-subscriptions-purchase-subscription-product.spec.js (54.065s)
  ● Subscriptions > Purchase subscription product › should be able to purchase the subscription product

    TimeoutError: Element td.subscription-next-payment (text: "October 30, 2021") not found

      waiting for function failed: timeout 500ms exceeded


Test Suites: 1 failed, 30 passed, 31 total
Tests:       1 failed, 84 passed, 85 total
Snapshots:   0 total
Time:        2218.978s
Ran all test suites.
```

**Review by naman03malhotra — APPROVED** [review]
LGTM! 🚢 

Though it would be great if we can get all the checks to pass before merging.

**Comment by htdat** [comment]
Latest commit: 36654ac was failed with the same reason  https://github.com/Automattic/woocommerce-payments/pull/2945#issuecomment-930884421

This E2E test will be fixed in #3016

**Comment by htdat** [comment]
Another merge commit: 23e3c18 was failed with the same reason  https://github.com/Automattic/woocommerce-payments/pull/2945#issuecomment-930884421

I've merged it so that it can run tests before we create a test package next Monday.

**Pull request merged**
